src/seedu/addressbook/commands/DisplayWeekCommand.java

```diff
@@ -0,0 +1,49 @@
+package seedu.addressbook.commands;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.Calendar;
+        
+/**
+ * Displays current week of the semester based on today's date.
+ */
+public class DisplayWeekCommand extends Command {
+
+    public static final String COMMAND_WORD = "displayweek";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ":\n" + "Displays the current week of the semester based on today's date.\n\t"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_DISPLAY_CURRENT_WEEK = "You are in %s of %s, %s";
+    public static final String MESSAGE_ACAD_CAL_NOT_FOUND = "Unable to find academic calendar file.";
+    public static final String MESSAGE_ACAD_CAL_READ_ERROR = "Unable to read academic calendar file.";
+
+    @Override
+    public CommandResult execute() {
+        try {
+            String filePath = "academiccalendar.txt";
```

Comment:
"Magic strings" should be avoided. Could be a final variable called WEEK_FILE_PATH (though may not be the best way to implement the feature, as described in the main comment)